### DISCUSSION TITLE
How to Configure NPMPLUS to Reverse Proxy Itself?

### DISCUSSION THREAD

**Discussion by ansible34** [ZoeyVid/NPMplus#1229] [Q&A] [answered]
Hi,

I’m trying to set up NPMPLUS to reverse proxy itself so that I can access the web interface securely via a custom domain name (e.g., https://npm-plus.mydomain.com). However, I’m having trouble getting this configuration to work correctly.

I’ve tried setting up a proxy host with the following settings:

    Scheme: http
    Forward Hostname/IP: 127.0.0.1
    Forward Port: 81 (since the web interface runs on port 81)
    SSL: Enabled using my wildcard certificate

Despite these settings, I keep encountering SSL warnings or issues with the web interface not loading properly. Could you please provide guidance or best practices on how to configure NPMPLUS to proxy itself effectively?

Thank you for your help!

Let me know if this captures what you want to ask or if you'd like to modify anything.

**Comment by Zoey2936** [comment, marked as answer]
you need to use https as Scheme

**Comment by ansible34** [comment]
on port 81 ?

**Reply by Zoey2936** [reply]
yes

**Comment by ansible34** [comment]
I tried using the wildcard certificate that I made with DNS challenge nothing works on the docker network and in host mode

**Reply by Zoey2936** [reply]
you need to use https as shema

**Comment by ansible34** [comment]
this is what I do before using this fork I use nginx-proxy-manager and it works

**Reply by Zoey2936** [reply]
did you changed the scheme to https?

**Comment by ansible34** [comment]
yes

**Reply by Zoey2936** [reply]
can you send me screenshots of your config?

**Comment by drajabr** [comment]
As someone coming from NPM and many many other reverse proxy gui's, this was frustrating to debug.

@Zoey2936 Probably need to be highlight a note in docs? Probably right after the part about the default password.

**Reply by Zoey2936** [reply]
I've added it to the migration part

**Comment by shadowjig** [comment]
![npmp-error](https://github.com/user-attachments/assets/16fc54c5-38f0-42ca-b2a9-41e4eb4a48c2)
I have the same issue.  I'm running the latest version (2.12.3+640668a).  When I configure NPMP with https scheme and address 127.0.0.1 and port 81.  Pick wildcard or fetch a cert for the hostname and enable force HTTPS, it fails to load in browser properly.
![npmp-config](https://github.com/user-attachments/assets/3eb1420e-3d4a-4adf-9fc7-9b82d80ddfc7)
![npmp-not-secure](https://github.com/user-attachments/assets/116dc73f-9331-4c39-8efe-e41df3824794)

**Reply by Zoey2936** [reply]
Thats because you are opening it on port 81 instead of 443 in your browser

**Reply by shadowjig** [reply]
If I remove 81 or change it to 443, the proxy overrides it back to 81.

**Reply by Zoey2936** [reply]
can you try in a different browser?

**Reply by shadowjig** [reply]
Works on Edge but not Chrome

**Reply by Zoey2936** [reply]
please try to clear the cache

**Comment by shadowjig** [comment]
OMG, did that last night and it didn't fix it.  But clearing cache in Chrome today did.  Thank you.
